### PR TITLE
chore: raise error when there is a multiplatform build without container registry

### PR DESCRIPTION
**Pull request by renzodavid9**
Fixes: #7633 
**Related**: #7082

**Description**

Due to the lack of support to build multi-platform images using the local Docker daemon, with this PR we add an actionable error when we detect a multi-platform build that does not specifies a container registry.

### CODE REVIEW

**Review by tejal29 — APPROVED** [review]
LGTM!

File: pkg/skaffold/build/errors.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+func noRegistryForMultiplatformBuildErr(err error) error {
+	return sErrors.NewError(err,
+		&proto.ActionableErr{
+			Message: err.Error(),
+			ErrCode: proto.StatusCode_BUILD_CROSS_PLATFORM_NO_REGISTRY_ERR,
+			Suggestions: []*proto.Suggestion{
+				{
+					SuggestionCode: proto.SuggestionCode_SET_PUSH_AND_CONTAINER_REGISTRY,
+					Action:         "To run multi-platform builds, set --push to true and set a container registry with --default-registry or in the global config",
```
Comment (gsquared94) [inline pkg/skaffold/build/errors.go:32]: typo: `--default-repo` instead of `--default-registry`.

Comment (renzodavid9) [reply pkg/skaffold/build/errors.go]: Done!

File: pkg/skaffold/build/builder_mux.go [hunk 2]
```diff
@@ -128,6 +128,11 @@ func (b *BuilderMux) Build(ctx context.Context, out io.Writer, tags tag.ImageTag
 		return built, nil
 	}
 
+	err := checkMultiplatformHaveRegistry(b, artifacts, resolver)
+	if err != nil {
+		return nil, fmt.Errorf("%w", err)
```
Comment (gsquared94) [inline pkg/skaffold/build/builder_mux.go:133]: can you create an issue for why we're doing the single unwrapping in the `ShowAIError` function. And add a todo comment here:
```
	return nil, fmt.Errorf("%w", err) // TODO: remove error wrapping after fixing #9999
```

**Comment by gsquared94** [comment]
you can fix the merge conflict in the `enums.proto` file and rerun `hack/generate-proto.sh` to fix the others.

**Review by gsquared94 — APPROVED** [review]

**Pull request merged**